### PR TITLE
Create dynamic repositories name for testing

### PR DESCRIPTION
**Pull request by AlexeiVainshtein**

### CODE REVIEW

File: jfrog-cli/utils/tests/utils.go [hunk 1]
```diff
@@ -308,3 +316,84 @@ func GetBaseDir(setToParentDir bool) (baseDir string) {
...
+
+	err = ioutil.WriteFile(template, []byte(content), 0700)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	// Now that the content is good we need to write the output to a file.
+	return template, nil
+}
+
+func PreparePath(fileName string) (string, error) {
```
Comment (eyalbe4) [inline jfrog-cli/utils/tests/utils.go]: I suggest renaming the function to CreateSpec.

File: jfrog-cli/jfrog/artifactory_test.go [hunk 2]
```diff
@@ -2536,23 +2883,41 @@ func execCreateRepoRest(repoConfig, repoName string) {
...
 
+func createRandomReposName() {
+	timestamp := strconv.FormatInt(time.Now().Unix(), 10)
+	tests.Repo1 += "-" + timestamp
+	tests.Repo2 += "-" + timestamp
+	tests.VirtualRepo += "-" + timestamp
+	tests.LfsRepo += "-" + timestamp
+	tests.JcenterRemoteRepo += "-" + timestamp
+	tests.NpmLocalRepo += "-" + timestamp
+	tests.NpmRemoteRepo += "-" + timestamp
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/jfrog/artifactory_test.go]: Done

File: jfrog-cli/utils/tests/utils.go [hunk 3]
```diff
@@ -308,3 +316,84 @@ func GetBaseDir(setToParentDir bool) (baseDir string) {
...
+			return "", errorutils.CheckError(err)
+		}
+		destPath = filepath.Join(destPath, Temp)
+	}
+	err = os.MkdirAll(destPath, 0700)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	template := filepath.Join(destPath, filepath.Base(path))
+	log.Debug("Writing content information to:", template)
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/utils/tests/utils.go]: Done

File: jfrog-cli/utils/tests/utils.go [hunk 4]
```diff
@@ -308,3 +316,84 @@ func GetBaseDir(setToParentDir bool) (baseDir string) {
...
+		for authenticationName, authenticationValue := range getAuthenticationMap() {
+			content = bytes.Replace(content, []byte(authenticationName), []byte(authenticationValue), -1)
+		}
+	}
+
+	err = ioutil.WriteFile(template, []byte(content), 0700)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	// Now that the content is good we need to write the output to a file.
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/utils/tests/utils.go]: That was for me. Sorry. I removed it

File: jfrog-cli/utils/tests/utils.go [hunk 5]
```diff
@@ -308,3 +316,84 @@ func GetBaseDir(setToParentDir bool) (baseDir string) {
...
+	return template, nil
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/utils/tests/utils.go]: Done

File: jfrog-cli/utils/tests/utils.go (hunk 1 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/utils/tests/utils.go]: Done

File: jfrog-cli/utils/tests/utils.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/utils/tests/utils.go]: Done

File: jfrog-cli/utils/tests/utils.go [hunk 6]
```diff
@@ -308,3 +316,84 @@ func GetBaseDir(setToParentDir bool) (baseDir string) {
...
+		"${VirtualRepo}":       VirtualRepo,
+		"${LfsRepo}":           LfsRepo,
+		"${JcenterRemoteRepo}": JcenterRemoteRepo,
+		"${NpmLocalRepo}":      NpmLocalRepo,
+		"${NpmRemoteRepo}":     NpmRemoteRepo,
+		"${GoRepo}":            GoLocalRepo,
+	}
+}
+
+func ReplaceTemplateVariables(path, destPath string, replaceCredentials bool) (string, error) {
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/utils/tests/utils.go]: Done. Removed the bool

File: jfrog-cli/utils/tests/utils.go [hunk 7]
```diff
@@ -308,3 +316,84 @@ func GetBaseDir(setToParentDir bool) (baseDir string) {
...
+	return map[string]string{
+		"${REPO1}":             Repo1,
+		"${REPO2}":             Repo2,
+		"${VirtualRepo}":       VirtualRepo,
+		"${LfsRepo}":           LfsRepo,
+		"${JcenterRemoteRepo}": JcenterRemoteRepo,
+		"${NpmLocalRepo}":      NpmLocalRepo,
+		"${NpmRemoteRepo}":     NpmRemoteRepo,
+		"${GoRepo}":            GoLocalRepo,
+	}
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply jfrog-cli/utils/tests/utils.go]: Done

**Pull request merged**